lib/src/objective_c.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// This library defines [NativeType]s for some core Objective C types.
+
+import 'dart:ffi';
+
+/// The Objective C `NSObject` type.
+///
+/// `NSObject` is always passed around by pointer, so use `Pointer<ObjCObject>`.
+/// The `id` type also maps to `Pointer<ObjCObject>`.
+class ObjCObject extends Opaque {}
+
+/// The Objective C `Class` type.
+///
+/// `Class` is always passed around by pointer, so use `Pointer<ObjCClass>`.
+class ObjCClass extends ObjCObject {}
```

Comment:
Enabling subtyping Opaque subtypes requires changes in the SDK:

* CFE: pkg/vm/lib/transformations/ffi/definitions.dart
* analyzer: pkg/analyzer/lib/src/generated/ffi_verifier.dart